### ISSUE TITLE
The dpploss value is a negative value

### ISSUE THREAD

**Issue by yaokeepmoving** [azadis/LDDP#2]
In file [dpp_loss_layer.py](https://github.com/azadis/LDDP/blob/master/lib/dpp_loss_layer.py),
if program execute the following code block
```python
            if among_ims.shape[0] == 0:  # 119th line in  file dpp_loss_layer.py
                log_p_Y.append(0)
                keeps_Y.append({})
```
the dpploss value will be negative
```python
top[0].data[...] = -normalizer_Y*sum(log_p_Y)+normalizer_B*sum(log_p_B)  # 203th line in file dpp_loss_layer.py
```
I am wondering whether it is reasonable where a loss value is negative.

In order to present my confusion, I print some information modifying few code.
My some modified code is here:
```python

            if among_ims.shape[0] == 0:
                print('IoU_with_gt_all_MAP = {}'.format(IoU_with_gt_all_MAP))
                print('cfg.TRAIN.IoU_gt_thresh = {}'.format(cfg.TRAIN.IoU_gt_thresh))
                print('good_gt_overlap = {}'.format(good_gt_overlap))
                print('MAP_images = {}'.format(MAP_images))
                print('among_ims = {}'.format(among_ims))
                log_p_Y.append(0)
                keeps_Y.append({})

.
.
.

        loss = -normalizer_Y * sum(log_p_Y) + normalizer_B * sum(log_p_B)
        if loss < 0:
            print('loss = {}'.format(loss))
            print('normalizer_Y = {}, log_p_Y = {}'.format(normalizer_Y, log_p_Y))
            print('normalizer_B = {}, log_p_B = {}'.format(normalizer_B, log_p_B))
            sys.exit(0)

        top[0].data[...] = -normalizer_Y * sum(log_p_Y) + normalizer_B * sum(log_p_B)

```

And corresponding output is as following:
> IoU_with_gt_all_MAP = [ 0.31134339  0.35766231  0.35166206  0.06769374]
cfg.TRAIN.IoU_gt_thresh = 0.5
good_gt_overlap = []
MAP_images = [0 1 2 3]
among_ims = []
loss = -0.941861043754
normalizer_Y = 1.0, log_p_Y = [0]
normalizer_B = 0.00505050505051, log_p_B = [-186.48848666331085]

Maybe there exist some mistakes in my code or data.

**Comment by azadis** [comment]
It is fine here to have a negative value for loss function. The intuition behind this loss in maximizing p(Y|X) and minimizing p(B|X) simultaneously. Decreasing value of the loss (either positive or negative) meets this expectation.

**Comment by yaokeepmoving** [comment]
Your answer help me acquire a better understanding for LDDP-Net.Thanks a lot!
